framework/Distributions.py

```diff
@@ -2441,6 +2443,116 @@ def rvs(self):
 
 DistributionsCollection.addSub(Custom1D.getInputSpecification())
 
+class LogUniform(Distribution):
+  """
+    Log Uniform univariate distribution
+    If x~LogUnif(a,b) then log(x)~Unif(log(a),log(b))
+  """
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Method to get a reference to a class that specifies the input data for
+      class cls.
+      @ In, cls, the class for which we are retrieving the specification
+      @ Out, inputSpecification, InputData.ParameterInput, class to use for
+        specifying input of cls.
+    """
+    inputSpecification = super(LogUniform, cls).getInputSpecification()
+    inputSpecification.addSub(InputData.parameterInputFactory("lowerBound", contentType=InputData.FloatType))
+    inputSpecification.addSub(InputData.parameterInputFactory("upperBound", contentType=InputData.FloatType))
+    inputSpecification.addSub(InputData.parameterInputFactory("base"      , contentType=InputData.StringType))
```

Comment:
Instead of an ```InputData.StringType``` use 
Enumerator Type... in this way you can list ```['decimal','natural']``` here directly and not need to add a check below

See for an example of usage in ```ROM.py``` parameter ```coeffRegressorEnumType```